cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceRuntimeService.java

```diff
@@ -373,9 +373,30 @@ protected void startUp() throws Exception {
   protected void run() throws Exception {
     MapReduceMetricsWriter metricsWriter = new MapReduceMetricsWriter(job, context);
 
+    int reportInterval = cConf.getInt(Constants.AppFabric.MAPREDUCE_STATUS_REPORT_INTERVAL_SECONDS);
+    String val = context.getRuntimeArguments().get(Constants.AppFabric.MAPREDUCE_STATUS_REPORT_INTERVAL_SECONDS);
+    if (val != null) {
+      try {
+        int interval = Integer.parseInt(val);
+        if (interval < 1) {
+          throw new NumberFormatException("Must be at least 1.");
+        }
+        reportInterval = interval;
+      } catch (NumberFormatException e) {
+        LOG.warn("Invalid value '{}' for '{}' given in runtime arguments: {}. Using default of {} seconds.",
+                 val, Constants.AppFabric.MAPREDUCE_STATUS_REPORT_INTERVAL_SECONDS, e.getMessage(), reportInterval);
+      }
+    }
+    LOG.debug("Interval for reporting MapReduce stats is {} seconds.", reportInterval);
+    long reportIntervalMillis = TimeUnit.SECONDS.toMillis(reportInterval);
+    long nextTimeToReport = 0L;
+
     // until job is complete report stats
     while (!job.isComplete()) {
-      metricsWriter.reportStats();
+      if (System.currentTimeMillis() >= nextTimeToReport) {
+        metricsWriter.reportStats();
+        nextTimeToReport = System.currentTimeMillis() + reportIntervalMillis;
```

Comment:
Because we still want to poll for job completion every second. Otherwise it may take a minute after a job completes until the CDAP MapReduce completes.